modules/sagemaker/sagemaker-templates-service-catalog/templates/model_deploy/product_stack.py

```diff
@@ -110,102 +106,210 @@ def __init__(
+                        value=json.dumps(dev_security_group_ids)
+                    ),
+                    "PRE_PROD_VPC_ID": codebuild.BuildEnvironmentVariable(value=pre_prod_vpc_id),
+                    "PRE_PROD_ACCOUNT_ID": codebuild.BuildEnvironmentVariable(value=pre_prod_account_id),
+                    "PRE_PROD_REGION": codebuild.BuildEnvironmentVariable(value=pre_prod_region),
+                    "PRE_PROD_SUBNET_IDS": codebuild.BuildEnvironmentVariable(value=json.dumps(pre_prod_subnet_ids)),
+                    "PRE_PROD_SECURITY_GROUP_IDS": codebuild.BuildEnvironmentVariable(
+                        value=json.dumps(pre_prod_security_group_ids)
                     ),
-                    iam.PolicyStatement(
-                        actions=[
-                            "kms:Encrypt",
-                            "kms:Decrypt",
-                            "kms:ReEncrypt*",
-                            "kms:GenerateDataKey*",
-                            "kms:DescribeKey",
-                        ],
-                        resources=[
-                            "*",
-                        ],
-                        principals=[
-                            iam.AccountPrincipal(pre_prod_account_id),
-                            iam.AccountPrincipal(prod_account_id),
-                        ],
+                    "PROD_VPC_ID": codebuild.BuildEnvironmentVariable(value=prod_vpc_id),
+                    "PROD_ACCOUNT_ID": codebuild.BuildEnvironmentVariable(value=prod_account_id),
+                    "PROD_REGION": codebuild.BuildEnvironmentVariable(value=prod_region),
+                    "PROD_SUBNET_IDS": codebuild.BuildEnvironmentVariable(value=json.dumps(prod_subnet_ids)),
+                    "PROD_SECURITY_GROUP_IDS": codebuild.BuildEnvironmentVariable(
+                        value=json.dumps(prod_security_group_ids)
                     ),
-                ]
+                },
             ),
         )
-
-        pipeline_artifact_bucket = s3.Bucket(
-            self,
-            "PipelineBucket",
-            bucket_name=f"pipeline-{sagemaker_project_name}-{sagemaker_project_id}-{Aws.ACCOUNT_ID}",
-            encryption_key=kms_key,
-            versioned=True,
-            removal_policy=RemovalPolicy.DESTROY,
-        )
-        # Block non-SSL
-        pipeline_artifact_bucket.add_to_resource_policy(
-            iam.PolicyStatement(
-                sid="AllowSSLOnly",
-                actions=["s3:*"],
-                effect=iam.Effect.DENY,
-                resources=[
-                    pipeline_artifact_bucket.bucket_arn,
-                    pipeline_artifact_bucket.arn_for_objects(key_pattern="*"),
-                ],
-                conditions={"Bool": {"aws:SecureTransport": "false"}},
-                principals=[iam.AnyPrincipal()],
+        # Verify that the project.role is not None
+        if project.role is None:
```

Comment:
I think we can skip this check if we can enforce the dependency between the resources (i.e. CodeBuild and it's role is created before the policy is attached).